MAVProxy/modules/mavproxy_link.py

```diff
@@ -208,6 +207,11 @@ def link_add(self, descriptor, force_connected=False):
         '''add new link'''
         try:
             (device, optional_attributes) = self.parse_link_descriptor(descriptor)
+            # if there's only 1 colon in device, it must be serialdevice:baud
+            # As all network connections will be pre-pended with tcp:, udpin:, etc
```

Comment:
That's not the case; `localhost:1234` is the same as `udpin:localhost:1234`

Reply:
Ahh, did not did consider that. Have fixed up such to check for that case